scipy/stats/_entropy.py

```diff
@@ -163,40 +205,109 @@ def differential_entropy(
 
     """
     values = np.asarray(values)
-    values = np.moveaxis(values, axis, 0)
+    values = np.moveaxis(values, axis, -1)
+    n = values.shape[-1]  # number of observations
 
     if window_length is None:
-        window_length = math.floor(math.sqrt(len(values)) + 0.5)
+        window_length = math.floor(math.sqrt(n) + 0.5)
 
-    if not 2 <= 2 * window_length < len(values):
+    if not 2 <= 2 * window_length < n:
         raise ValueError(
             f"Window length ({window_length}) must be positive and less "
-            f"than half the sample size ({len(values)}).",
+            f"than half the sample size ({n}).",
         )
 
-    sorted_data = np.sort(values, axis=0)
+    if base is not None and base <= 0:
+        raise ValueError("`base` must be a positive number or `None`.")
 
-    repeats = np.array(
-        (window_length + 1,)
-        + ((1,) * (len(sorted_data) - 2))
-        + (window_length + 1,),
-    )
+    sorted_data = np.sort(values, axis=-1)
 
-    padded_data = np.repeat(
-        sorted_data,
-        repeats=repeats,
-        axis=0,
-    )
+    methods = {"vasicek": _vasicek_entropy,
+               "van es": _van_es_entropy,
+               "correa": _correa_entropy,
+               "ebrahimi": _ebrahimi_entropy,
+               "auto": None}
```

Comment:
```suggestion
               "auto": _vasicek_entropy}  # only used if N > 1000
```

Reply:
Maybe you could define it like `methods: Dict[str, Optional[callable]] = ...` or just skip mypy check here. Does not look good to put a wrong thing here.

Reply:
It's not necessarily wrong. For instance, we could remove the 
```
        else:
            method = 'vasicek'
```
below and then it would actually be used as part of the logic. If n > 1000, `method` would remaing `auto` and then `methods[method]` would return `_vasicek_entropy`. The would be equivalent.

That said, I would prefer to leave the `else` condition.

Re defining `methods` differently - could you click the `+` mark where that line should be inserted, and click the 
![image](https://user-images.githubusercontent.com/6570539/114588816-f0aff900-9c3b-11eb-91c9-538e107c2a57.png)

button in the comment window to add a specific suggestion to the code?

Reply:
I would also keep the `else` to keep it clear. It's obfuscated otherwise.